src/__tests__/get-user-code-frame.js

```diff
@@ -0,0 +1,91 @@
+import fs from 'fs'
+import {getUserCodeFrame} from '../get-user-code-frame'
+
+jest.mock('fs', () => ({
+  // We setup the contents of a sample file
+  readFileSync: jest.fn(
+    () => `
+    import {screen} from '@testing-library/dom'
+    it('renders', () => {
+      document.body.appendChild(
+        document.createTextNode('Hello world')
+      )
+      screen.debug()
+      expect(screen.getByText('Hello world')).toBeInTheDocument()
+    })
+  `,
+  ),
+}))
+
+const userStackFrame =
+  'at somethingWrong (/home/john/projects/sample-error/error-example.js:7:14)'
+
+let globalErrorMock
+
+beforeEach(() => {
+  // Mock global.Error so we can setup our own stack messages
+  globalErrorMock = jest.spyOn(global, 'Error')
+})
+
+afterEach(() => {
+  global.Error.mockRestore()
+})
+
+test('it returns only client code frame when code frames from node_modules are first', () => {
+  const stack = `Error: Kaboom
+      at Object.<anonymous> (/home/john/projects/projects/sample-error/node_modules/@es2050/console/build/index.js:4:10)
```

Comment:
This says `john/projects/projects`, but the snapshot just says `john/projects/` 🤔

Reply:
On this particular test, that frame should actually be ignored since it's inside node_modules. The one we care about is one line after `${userStackFrame}`.
However, the frame that is ignored does have a typo (`/projects/projects`). I've corrected this and simplified the paths used for all tests to avoid confusion. They all now start with `/sample-error` instead of `/home/john/<etc>`